diff --git a/CMakeLists.txt b/CMakeLists.txt
index 727ba42..e86ca43 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -273,7 +273,7 @@ if(CLIENTS OR TESTS)
   set(REF_LIBRARIES )
   if(CLBLAS_FOUND)
     set(REF_INCLUDES ${REF_INCLUDES} ${CLBLAS_INCLUDE_DIRS})
-    set(REF_LIBRARIES ${REF_LIBRARIES} ${CLBLAS_LIBRARIES})
+    set(REF_LIBRARIES ${REF_LIBRARIES} ${CLBLAS_LIBRARIES} pthread)
     if(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
       add_definitions(" /DCLBLAST_REF_CLBLAS")
     else()
