diff --git a/modules/calib3d/src/ap3p.cpp b/modules/calib3d/src/ap3p.cpp
index 6aa771d..f7b27a5 100644
--- a/modules/calib3d/src/ap3p.cpp
+++ b/modules/calib3d/src/ap3p.cpp
@@ -35,7 +35,11 @@ void solveQuartic(const double *factors, double *realRoots) {
     else
         w = sqrt(static_cast<complex<double> >(q3 * q3 - p3 * p3 * p3)) - q3;
     if (w.imag() == 0.0) {
+#if __cplusplus >= 201103L
         w.real(cbrt(w.real()));
+#else
+        w.real() = cbrt(w.real());
+#endif
         t = 2.0 * (w.real() + p3 / w.real());
     } else {
         w = pow(w, 1.0 / 3);
